nupic/research/frameworks/dendrites/routing/learn.py

```diff
@@ -70,30 +86,39 @@ def init_test_scenario(dim_in, dim_out, num_contexts, dim_context, dendrite_modu
         percent_on=0.2
     )
 
-    # Initialize dendrite module using the same feed-forward sparse weights as the
-    # routing function; also note that the value passed to `dendrite_sparsity` is
-    # irrelevant since the context weights are subsequently overwritten
-    dendritic_network = dendrite_module(
-        module=r.sparse_weights.module,
+    # If only training the dendrite weights, initialize the dendrite module using the
+    # same feed-forward sparse weights as the routing function, otherwise if learning
+    # feed-forward weights, use `torch.nn.Linear`
+
+    # Also, note that the value passed to `dendrite_sparsity` is irrelevant since the
+    # context weights are subsequently
+    # overwritten
+    if mode == "dendrites":
+        dendrite_layer_forward_module = r.sparse_weights.module
+    elif mode == "all":
```

Comment:
Should probably throw an exception in the else clause, or otherwise ensure mode is not set to something else.

Reply:
Good point. I added in the exception clause just here, and nowhere else where there's an if statement around `mode` since this is the first time such as if statement is performed.